sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -1382,6 +1384,8 @@ abstract Builder<T> setFormatFn(
 
       abstract Builder<T> setPubsubRootUrl(String pubsubRootUrl);
 
+      abstract Builder<T> setNeedsOrderingKey(boolean needsOrderingKey);
```

Comment:
needs still seems confusing as verb to me for output. For input it makes since because the consumer requested ordering id and thus we need to send it.  But for output "needs" sounds like it requires the incoming messages to have an ordering id set which isn't true.

Can we change it to setSupportsOrderingKey or setPublishWithOrderingKey?

Reply:
That's fair, switched it to `publishWithOrderingKey`